package edu.colorado.plv.fixr.protobuf;

import "proto_acdfg.proto";

message Lattice {
  message AcdfgBin {
    required uint64 id = 1;
    required Acdfg acdfg_repr = 2;
    repeated string acdfg_names = 3;
    repeated uint64 subsuming_bins = 4;
    repeated uint64 immediate_subsuming_bins = 5;
    repeated uint64 incoming_edges = 6;
    required bool subsuming = 7;
    required bool anomalous = 8;
    required bool popular = 9;
  }

  repeated AcdfgBin bins = 1;
  repeated AcdfgBin popular_bins = 2;
  repeated AcdfgBin anomalous_bins = 3;
  repeated AcdfgBin isolated_bins = 4;
}
